backend/data/processors/programs_processing.py

```diff
@@ -139,7 +154,11 @@ def addDisciplineData(components, item):
                     ):
                         code = major["academic_item_code"]
                         majorData[code] = major["academic_item_name"]
-                SpecialisationData["Majors"] = majorData
+                try:
+                    components["SpecialisationData"]["Majors"].update({programName: majorData})
+                except KeyError:
+                    components["SpecialisationData"]["Majors"] = {programName: majorData}
```

Comment:
```suggestion
                    components["SpecialisationData"].setdefault("Majors", {}).update({programName: majorData})
```
I think this performs the same and would make this pattern much neater

Reply:
That's neat! Thank you 🤩